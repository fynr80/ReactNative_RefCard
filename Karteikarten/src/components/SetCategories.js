/**
 * @file Get and Return Categories & Style for HomeScreen
 * @author Fatih Yanar fatih_yanar@gmx.de
 */

import * as React from 'react';
import {
    View,
    Text,
    TouchableOpacity,
    ImageBackground
} from "react-native";
import {useEffect, useState} from "react";

//import {LinearGradient} from 'expo-linear-gradient';
import firebase from "firebase/compat";
import {useNavigation} from '@react-navigation/native';
import IconF from 'react-native-vector-icons/FontAwesome'

import stylejs from "../style";


const SetCategories = (props) => {
    const id = props.id
    const categoriesName = props.name
    const categoryColor = props.categoryColor;
    const navigation = useNavigation();
    const [themeCounter, setThemeCounter] = useState(0);


    useEffect(() => {
        return getThemeCounter()
    })

    /**
     * Get the amount of Themes of each Category
     */
    const getThemeCounter = () => {
        firebase.firestore()
            .collection('category')
            .doc(props.id)
            .collection("set")
            .onSnapshot(snapshot => {
                const arr = snapshot.docs.map((doc) => ({
                    id: doc.id,
                    name: doc.data().titel
                }))
                setThemeCounter(arr.length)
            })
    }

    /**
     * Got the SearchScreen of the selected Category
     */
    const goToSearch = () => {
        navigation.navigate("Search", {id, categoriesName, categoryColor})
    }

    return (
        <TouchableOpacity
            onPress={goToSearch}
        >
            <View testID="homescreenSet" style={[stylejs.items_Container,{height:120, backgroundColor: categoryColor.left}]}>
            {/* <LinearGradient style={[stylejs.items_Container,{height:120}]} colors={[categoryColor.left,categoryColor.right]}>*/}
                <ImageBackground  source={require('../asserts/intelligence.png')} imageStyle={{width:'30%',position:'absolute',left:100,opacity:0.3}} style={{width: '100%', height: '100%',flexDirection:'row',justifyContent:'space-between'}}>
                    <View >
                        <Text testID={"homescreenSetTitle"} numberOfLines={2} style={stylejs.items_titleText}>{props.name}</Text>
                        <Text style={stylejs.items_amountText}>{themeCounter} Sets</Text>
                    </View>
                    <View>
                        <IconF name='chevron-right' size={35} color='white'/>
                    </View>
                </ImageBackground>
            {/*  </LinearGradient>*/}
            </View>
        </TouchableOpacity>
    )
};

export default SetCategories;
