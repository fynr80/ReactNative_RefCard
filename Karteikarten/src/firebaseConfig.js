
import firebase from 'firebase/compat/app';

const firebaseConfig = {
    apiKey: "AIzaSyBksswFLr1YB6en-sRf8mqSFle7bhAzSgU",
    authDomain: "swtp-812df.firebaseapp.com",
    databaseURL: "https://swtp-812df-default-rtdb.firebaseio.com",
    projectId: "swtp-812df",
    storageBucket: "swtp-812df.appspot.com",
    messagingSenderId: "44236558384",
    appId: "1:44236558384:web:2f15f6b5fb190634b71df5",
    measurementId: "G-W33L0G7D18"
};
firebase.initializeApp(firebaseConfig);
export {firebase};
