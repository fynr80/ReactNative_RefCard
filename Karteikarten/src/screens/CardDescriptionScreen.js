/**
 * @file CardDescriptionScreen for RefCard
 * @author Ibrahim Eraslan <ibrahim.eraslan@mni.thm.de>
 */


import * as React from 'react';
import {
    StyleSheet,
    View,
    Text,
    TouchableOpacity
} from 'react-native';

import Icon from 'react-native-vector-icons/Ionicons'
import {SafeAreaView} from 'react-native-safe-area-context';
//import {LinearGradient} from "expo-linear-gradient";
import {useNavigation} from "@react-navigation/native";

import stylejs from "../style";
import SetDropDown from "../components/SetDropDown";


export default function CardDescriptionScreen(props) {
    const itemName = props.route.params.itemName;
    const item_description = props.route.params.item_description;
    const categoryId = props.route.params.category_id;
    const item_id = props.route.params.item_id;
    const categoryColor = props.route.params.categoryColor;
    const questionCounter = props.route.params.questionCounter

    const navigation = useNavigation();

    return (
        <SafeAreaView style={styles.container}>
            <View style={styles.header}>
                <TouchableOpacity
                    testID="CardDescriptionbackbutton"
                    onPress={() => {
                        navigation.goBack();
                    }}>
                    <Icon name='chevron-back-sharp' size={50} color='#FD4365'/>
                </TouchableOpacity>

                <Text testID="cardDescriptionHeadline" style={styles.headerText}>DESCRIPTION</Text>
                <SetDropDown style={stylejs.items_dropdown}
                             title={props.text}
                             item_id={item_id}
                             categorieID={categoryId}/>
            </View>
            <View style={styles.topMain}>
                <View testID="a" style={[styles.cardDescription, {backgroundColor: categoryColor.left}]}>
                    {/*<LinearGradient colors={[categoryColor.left, categoryColor.right]} style ={styles.cardDescription}>*/}
                    <View>
                        <Text
                            testID="cardDescriptionItemName"
                            style={{
                            color: 'white',
                            textAlign: 'center',
                            fontWeight: 'bold',
                            fontSize: 20
                        }}>{itemName}</Text>
                        <Text testID="cardDescriptionItemDescription" style={{color: 'white', textAlign: 'left', marginTop: 5}}>{item_description}</Text>
                    </View>
                    <View style={styles.underTopDescription}>
                        <View>
                            <Text testID="cardDescriptionQuestionCounter" style={{color: 'white', textAlign: 'center', marginTop: 5}}>{questionCounter}</Text>
                            <Text style={{color: 'white', textAlign: 'left'}}>Questions</Text>
                        </View>
                        <View>
                            <Text testID="cardDescriptionQuestionCounter" style={{color: 'white', textAlign: 'center', marginTop: 5}}>{questionCounter}</Text>
                            <Text style={{color: 'white', textAlign: 'left'}}>Questions</Text>
                        </View>
                    </View>
                    {/*</LinearGradient>*/}
                </View>
            </View>
            <View style={styles.underMain}>
                <TouchableOpacity activeOpacity={.8}
                                  style={styles.topUnderMain}
                                  testID="Learn-mode"
                                  onPress={() => {
                                      navigation.navigate('CardQuery', {
                                          category_id: categoryId,
                                          item_id,
                                          categoryColor
                                      });
                                  }}>
                    <View style={styles.button}>
                        <View style={styles.buttonText}>
                            <Text style={{
                                color: 'black',
                                textAlign: 'left',
                                fontWeight: 'bold',
                                fontSize: 20
                            }}>Learn-mode</Text>
                            <Text style={{color: 'black', textAlign: 'left', fontSize: 15}}>Learning according to Spaced
                                Repetition</Text>
                        </View>
                        <View>
                            <Icon name='play' size={40} color='black'/>
                        </View>
                    </View>
                </TouchableOpacity>
            </View>
        </SafeAreaView>
    );
}


const styles = StyleSheet.create({
    container: {        
        flex: 1,
        justifyContent: 'center',
        alignContent: 'flex-start',

    },
    header: {
        backgroundColor: 'white',
        flexDirection: 'row',
        flex: 1,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingLeft: 2,
        paddingRight: 2,
    },
    headerText: {
        fontSize: 20,
        fontWeight: 'bold'
    },
    cardDescription: {
        backgroundColor: '#FE357D',
        width: '90%',
        height: '90%',
        borderRadius: 26,
        padding: 10,
        justifyContent: 'space-between',
        paddingBottom: 30,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 3,
        },
        shadowOpacity: 0.27,
        shadowRadius: 4.65,
        elevation: 6,
    },
    underTopDescription: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        paddingLeft: 30,
        paddingRight: 30,
    },
    topMain: {
        backgroundColor: 'white',
        flex: 5,
        justifyContent: 'center',
        alignItems: 'center'
    },
    underMain: {
        backgroundColor: 'white',
        flex: 5,
        justifyContent: 'center'
    },
    topUnderMain: {
        justifyContent: 'center',
        padding: 10,
    },
    button: {
        backgroundColor: '#FBF9F9',
        alignItems: 'center',
        justifyContent: 'space-between',
        flexDirection: 'row',
        borderRadius: 18,
        padding: 10,
        height: 90,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 3,
        },
        shadowOpacity: 0.27,
        shadowRadius: 4.65,
        elevation: 6,
    },
});

