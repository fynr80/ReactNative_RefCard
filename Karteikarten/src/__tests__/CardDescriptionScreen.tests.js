import React from 'react';
import {fireEvent, render} from '@testing-library/react-native'
import CardDescriptionScreen from "../screens/CardDescriptionScreen";
import {NavigationContainer} from "@react-navigation/native";
import firebaseConfig from "../firebaseConfig";
import SetCategories from "../components/SetCategories";
import {toMatchImageSnapshot} from "jest-image-snapshot";

const puppeteer = require("puppeteer")

const mockedNavigation = jest.fn();
jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: mockedNavigation,
            dispatch: jest.fn(),
        }),
    };
});
const route = {
    params: {
        itemName: "math",
        item_description: "Set for math",
        category_id: "2",
        item_id: "2.2",
        categoryColor: "",
        questionCounter: 0
    }
}

expect.extend({toMatchImageSnapshot});

describe("CardDescriptionScreen Components", () => {
    let browser;
    jest.setTimeout(35000);

    beforeAll(async () => {
        browser = await puppeteer.launch({args: ['--no-sandbox', '--disable-setuid-sandbox']});
    })

    it('renders correctly', () => {
        const tree = render(<CardDescriptionScreen route={route} navigation={{mockedNavigation}}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    /*it('getting back', () => {
        const {getByTestId} = render(
            <NavigationContainer>
                <CardDescriptionScreen route={route} navigation={{mockedNavigation}}/>
            </NavigationContainer>
        )
        const backbutton = getByTestId("CardDescriptionbackbutton");
        //fireEvent.press(backbutton);
    });*/
    it("it makes a screenshot", async () => {
        const page = await browser.newPage();
        await page.goto('http://localhost:19006');

        const clickSetSelector = "#animatedComponent > div > div > div > div.css-view-1dbjc4n.r-alignItems-1awozwy.r-backgroundColor-14lw9ot.r-flex-1q4bek2.r-flexDirection-eqz5dr.r-justifyContent-1wtj0ep.r-paddingLeft-1hfyk0a.r-paddingRight-1qfoi16 > div > div > div > div:nth-child(1) > div";
        await page.waitForSelector(clickSetSelector);
        const clickSet = await page.$(clickSetSelector);
        clickSet.click();

        const clickCardSelector = "#root > div > div > div > div > div > div.css-view-1dbjc4n.r-flex-13awgt0 > div.css-view-1dbjc4n.r-bottom-1p0dtai.r-display-6koalj.r-left-1d2f490.r-pointerEvents-12vffkv.r-position-u8s1d.r-right-zchlnj.r-top-ipm5af > div.css-view-1dbjc4n.r-bottom-1p0dtai.r-display-6koalj.r-left-1d2f490.r-pointerEvents-12vffkv.r-position-u8s1d.r-right-zchlnj.r-top-ipm5af > div.css-view-1dbjc4n.r-flex-13awgt0.r-pointerEvents-12vffkv > div > div > div > div.css-view-1dbjc4n.r-flex-13awgt0 > div > div:nth-child(3) > div > div > div > div";
        await page.waitForSelector(clickCardSelector);
        const clickCard = await page.$(clickCardSelector);
        clickCard.click();

        await new Promise((r) => setTimeout(r, 5000));

        const image = await page.screenshot();

        expect(image).toMatchImageSnapshot();
    })

    it('Navigate to Learnmode', () => {
        const {getByTestId} = render(<CardDescriptionScreen route={route} navigation={{mockedNavigation}}/>);
        const learnmodeButton = getByTestId("Learn-mode");
        fireEvent.press(learnmodeButton)
        expect(mockedNavigation).toBeCalledWith('CardQuery', {
                "category_id": "2",
                "item_id": "2.2",
                "categoryColor": "",
            }
        )

    });
    it('Correct headline', () => {
        const {getByTestId} = render(<CardDescriptionScreen route={route} navigation={{mockedNavigation}}/>)
        const cardDescriptionHeadline = getByTestId("cardDescriptionHeadline");
        expect(cardDescriptionHeadline.props.children).toEqual("DESCRIPTION")
    });
    it('Correct ItemValues', async () => {
        const {getByTestId, getAllByTestId} = await render(<CardDescriptionScreen route={route}
                                                                                  navigation={{mockedNavigation}}/>)
        const cardDescriptionItemName = getByTestId("cardDescriptionItemName");
        const cardDescriptionItemDescription = getByTestId("cardDescriptionItemDescription");
        // const cardDescriptionQuestionCounter = getAllByTestId("cardDescriptionQuestionCounter");
        expect(cardDescriptionItemName.props.children).toEqual("math");
        expect(cardDescriptionItemDescription.props.children).toEqual("Set for math");
        //expect(cardDescriptionQuestionCounter.props.children).toEqual(0);

    });

});
