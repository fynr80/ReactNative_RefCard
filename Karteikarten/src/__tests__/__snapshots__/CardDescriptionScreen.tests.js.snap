// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDescriptionScreen Components renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignContent": "flex-start",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingLeft": 2,
        "paddingRight": 2,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="CardDescriptionbackbutton"
    >
      <Text />
    </View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
      testID="cardDescriptionHeadline"
    >
      DESCRIPTION
    </Text>
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text />
        <View>
          <Modal
            animationType="none"
            deviceHeight={null}
            deviceWidth={null}
            hardwareAccelerated={false}
            hideModalContentWhileAnimating={false}
            onBackdropPress={[Function]}
            onModalHide={[Function]}
            onModalWillHide={[Function]}
            onModalWillShow={[Function]}
            onRequestClose={[Function]}
            panResponderThreshold={4}
            scrollHorizontal={false}
            scrollOffset={0}
            scrollOffsetMax={0}
            scrollTo={null}
            statusBarTranslucent={false}
            supportedOrientations={
              Array [
                "portrait",
                "landscape",
              ]
            }
            swipeThreshold={100}
            transparent={true}
            visible={false}
          >
            <View
              accessible={true}
              focusable={true}
              forwardedRef={[Function]}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "black",
                  "bottom": 0,
                  "height": 1334,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 750,
                }
              }
            />
            <View
              deviceHeight={null}
              deviceWidth={null}
              forwardedRef={[Function]}
              hideModalContentWhileAnimating={false}
              onBackdropPress={[Function]}
              onModalHide={[Function]}
              onModalWillHide={[Function]}
              onModalWillShow={[Function]}
              panResponderThreshold={4}
              pointerEvents="box-none"
              scrollHorizontal={false}
              scrollOffset={0}
              scrollOffsetMax={0}
              scrollTo={null}
              statusBarTranslucent={false}
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "margin": 0,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                  ],
                }
              }
              supportedOrientations={
                Array [
                  "portrait",
                  "landscape",
                ]
              }
              swipeThreshold={100}
            >
              <View>
                <RCTScrollView
                  ListFooterComponent={[Function]}
                  ListHeaderComponent={[Function]}
                  data={
                    Array [
                      "Edit",
                      "Share",
                      "Delete",
                    ]
                  }
                  disableVirtualization={false}
                  getItem={[Function]}
                  getItemCount={[Function]}
                  horizontal={false}
                  initialNumToRender={10}
                  keyExtractor={[Function]}
                  maxToRenderPerBatch={10}
                  onContentSizeChange={[Function]}
                  onEndReachedThreshold={2}
                  onLayout={[Function]}
                  onMomentumScrollEnd={[Function]}
                  onScroll={[Function]}
                  onScrollBeginDrag={[Function]}
                  onScrollEndDrag={[Function]}
                  removeClippedSubviews={false}
                  renderItem={[Function]}
                  scrollEnabled={false}
                  scrollEventThrottle={50}
                  stickyHeaderIndices={Array []}
                  updateCellsBatchingPeriod={50}
                  viewabilityConfigCallbackPairs={Array []}
                  windowSize={21}
                >
                  <View>
                    <View
                      onLayout={[Function]}
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "backgroundColor": "#e6e6e6",
                              "borderTopLeftRadius": 20,
                              "borderTopRightRadius": 20,
                              "padding": "2%",
                            }
                          }
                        />
                      </View>
                    </View>
                    <View
                      onLayout={[Function]}
                      style={null}
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#e6e6e6",
                            "padding": "1%",
                          }
                        }
                      >
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "borderRadius": 20,
                                "padding": "1.5%",
                              },
                              Object {
                                "backgroundColor": "white",
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Object {
                                "fontSize": 20,
                                "fontWeight": "bold",
                                "textAlign": "center",
                              }
                            }
                          >
                            Edit
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      onLayout={[Function]}
                      style={null}
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#e6e6e6",
                            "padding": "1%",
                          }
                        }
                      >
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "borderRadius": 20,
                                "padding": "1.5%",
                              },
                              Object {
                                "backgroundColor": "white",
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Object {
                                "fontSize": 20,
                                "fontWeight": "bold",
                                "textAlign": "center",
                              }
                            }
                          >
                            Share
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      onLayout={[Function]}
                      style={null}
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#e6e6e6",
                            "padding": "1%",
                          }
                        }
                      >
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "borderRadius": 20,
                                "padding": "1.5%",
                              },
                              Object {
                                "backgroundColor": "white",
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Object {
                                "fontSize": 20,
                                "fontWeight": "bold",
                                "textAlign": "center",
                              }
                            }
                          >
                            Delete
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      onLayout={[Function]}
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "backgroundColor": "#e6e6e6",
                              "padding": "4%",
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
              </View>
            </View>
          </Modal>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 5,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FE357D",
            "borderRadius": 26,
            "elevation": 6,
            "height": "90%",
            "justifyContent": "space-between",
            "padding": 10,
            "paddingBottom": 30,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.27,
            "shadowRadius": 4.65,
            "width": "90%",
          },
          Object {
            "backgroundColor": undefined,
          },
        ]
      }
      testID="a"
    >
      <View>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
          testID="cardDescriptionItemName"
        >
          math
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "marginTop": 5,
              "textAlign": "left",
            }
          }
          testID="cardDescriptionItemDescription"
        >
          Set for math
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 30,
            "paddingRight": 30,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "white",
                "marginTop": 5,
                "textAlign": "center",
              }
            }
            testID="cardDescriptionQuestionCounter"
          >
            0
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "textAlign": "left",
              }
            }
          >
            Questions
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "white",
                "marginTop": 5,
                "textAlign": "center",
              }
            }
            testID="cardDescriptionQuestionCounter"
          >
            0
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "textAlign": "left",
              }
            }
          >
            Questions
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 5,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "justifyContent": "center",
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="Learn-mode"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FBF9F9",
            "borderRadius": 18,
            "elevation": 6,
            "flexDirection": "row",
            "height": 90,
            "justifyContent": "space-between",
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.27,
            "shadowRadius": 4.65,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "left",
              }
            }
          >
            Learn-mode
          </Text>
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 15,
                "textAlign": "left",
              }
            }
          >
            Learning according to Spaced Repetition
          </Text>
        </View>
        <View>
          <Text />
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
